### PR TITLE
143534095 documentation

### PR DESCRIPTION
**Pull request by robbie-smith**
@njgheorghita @Sh1pley @icorson3 

closes #143534095

Documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,39 @@
...
+This app consists of two components, this is the back-end portion. The front-end
+is written in React, and is located here [Bookchain-react](https://github.com/Sh1pley/bookchain-react). This repository is written in Solidity and is built upon the block chain.
 
-### A decentralized virtual library to keep track of your centralized physical library
\ No newline at end of file
+#### Getting Started
+1. This project is built in Solidity, want more info? See the resources section
+   below.
+2. git clone or fork this repository `$ git clone https://github.com/njgheorghita/bookchain.git`
+3. Install Solidity `$ npm install -g solc`
```
Comment (njgheorghita) [inline README.md]: not sure that it's necessary to install the solidity compiler?? i think truffle takes care of that for us, but i could be wrong

File: README.md [hunk 2]
```diff
@@ -1,3 +1,39 @@
...
+4. Install Truffle `$ npm install -g truffle`
+5. Install Testrpc `$ npm install ethereumjs-testrpc`
+
+#### Running
+1. Run `$ truffle migrate` this will migrate the contracts
+
+#### Testing
+1. In a separate terminal session run `$ testrpc`
```
Comment (njgheorghita) [inline README.md]: you need to start up testrpc before migrating the contracts with truffle

File: README.md [hunk 3]
```diff
@@ -1,3 +1,39 @@
...
+
+*Running the command `testrpc`, runs the localhost blockchain on port 8545*
+
+2. Run `$ truffle test` this command will run the test suite.
+
+*You can also run `truffle test test/<filename>` to run a single test file
+instead of the entire suite*
+
+#### Resources
```
Comment (njgheorghita) [inline README.md]: this is tight

File: README.md [hunk 4]
```diff
@@ -1,3 +1,39 @@
...
+* [Solidity](https://solidity.readthedocs.io/en/develop/#)
+* [Truffle](http://truffleframework.com/docs/)
+* [Testrpc](https://github.com/ethereumjs/testrpc)
+* [Web3](https://github.com/ethereum/web3.j://github.com/ethereum/web3.js)
+* [Mocha](https://mochajs.org)
+* [Chai](http://chaijs.com)
+
+#### Contributions
+1. Fork the repository
+2. Find a problem
```
Comment (njgheorghita) [inline README.md]: shouldn't be too hard to find, haha

File: README.md (hunk 1 above)
Comment (robbie-smith) [reply README.md]: @njgheorghita I wasn't sure about that. To be honest I couldn't remember, when I added that I was referencing the Solidity docs.

File: README.md (hunk 2 above)
Comment (robbie-smith) [reply README.md]: @njgheorghita Cool I'll make that change.

File: README.md (hunk 1 above)
Comment (robbie-smith) [reply README.md]: @njgheorghita Changes made!

**Comment by icorson3** [comment]
@njgheorghita how about documenting endpoints and your api? maybe that is in a different repo but connect them 👍

**Comment by robbie-smith** [comment]
@icorson3 The api endpoints are through web3 and behind the scenes.

**Pull request merged**